### ISSUE TITLE
`list` of `dict` is reported as not compatible with `list` of `Mapping`

### ISSUE THREAD

**Issue by Czaki** [python/mypy#17752]
**Bug Report**

Then have a function that returns a list of dicts and assigns it to a variable annotated as list of Mapping the mypy reports error, however when change annotation to `Sequence[Mapping]` it works.  

**To Reproduce**

Code that generates mypy warning 

```python
from collections.abc import Mapping, Sequence


def get_dkt() -> list[dict]:
    return [{}]
    
var2: list[Mapping]

var2 = get_dkt()
```

code that do not generate error:

```python
from collections.abc import Mapping, Sequence


def get_dkt() -> list[dict]:
    return [{}]
    
var2: Sequence[Mapping]

var2 = get_dkt()
```

https://mypy-play.net/?mypy=latest&python=3.12&gist=a3a7cf085ceed0a395392a6adaa74a29

**Expected Behavior**

Both presented type annotation above should work.

**Actual Behavior**

**Your Environment**

- Mypy version used: 1.11.2
- Python version used: 3.12

**Comment by JelleZijlstra** [comment]
https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance
